ruby/ql/lib/codeql/ruby/frameworks/data/internal/ApiGraphModelsSpecific.qll

```diff
@@ -0,0 +1,141 @@
+/**
+ * Contains the language-specific part of the models-as-data implementation found in `ApiGraphModels.qll`.
+ *
+ * It must export the following members:
+ * ```codeql
+ * class Unit // a unit type
+ * class InvokeNode // a type representing an invocation connected to the API graph
+ * module API // the API graph module
+ * predicate isPackageUsed(string package)
+ * API::Node getExtraNodeFromPath(string package, string type, string path, int n)
+ * API::Node getExtraSuccessorFromNode(API::Node node, AccessPathToken token)
+ * API::Node getExtraSuccessorFromInvoke(InvokeNode node, AccessPathToken token)
+ * predicate invocationMatchesExtraCallSiteFilter(InvokeNode invoke, AccessPathToken token)
+ * InvokeNode getAnInvocationOf(API::Node node)
+ * ```
+ */
+
+private import ruby
+private import codeql.ruby.DataFlow
+private import codeql.ruby.dataflow.internal.DataFlowPrivate as DataFlowPrivate
+private import ApiGraphModels
+
+class Unit = DataFlowPrivate::Unit;
+
+import codeql.ruby.ApiGraphs
+import codeql.ruby.dataflow.internal.AccessPathSyntax as AccessPathSyntax
```

Comment:
Use `private import` if possible to avoid pulling all transitively imported names into scope.

```suggestion
private import codeql.ruby.ApiGraphs
private import codeql.ruby.dataflow.internal.AccessPathSyntax as AccessPathSyntax
```

Reply:
@hmac is right. I'll add a comment to explain, since deliberate re-export can be hard to distinguish from accidental re-export.